plugin/storage/integration/integration.go

```diff
@@ -144,6 +144,32 @@ func (s *StorageIntegration) testGetServices(t *testing.T) {
 	}
 }
 
+func (s *StorageIntegration) testArchiveTrace(t *testing.T) {
+	defer s.cleanUp(t)
+	tID := model.NewTraceID(uint64(11), uint64(22))
+	expected := &model.Span{
+		OperationName: "archive_span",
+		StartTime:     time.Now().Add(-time.Hour * 72 * 5),
+		TraceID:       tID,
+		SpanID:        model.NewSpanID(55),
+		References:    []model.SpanRef{},
+		Process:       model.NewProcess("archived_service", model.KeyValues{}),
+	}
+
+	require.NoError(t, s.SpanWriter.WriteSpan(context.Background(), expected))
```

Comment:
I suggest changing this to use `s.ArchiveSpanWriter` (to be initialized by the actual implementation), and to skip the test if that writer is nil

Reply:
Yes, Understood that. Already made that change.